src/index.js

```diff
@@ -3,8 +3,19 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
+import { Provider } from 'react-redux';
+import productReducer from './reducers/products';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+const store = createStore(productReducer, applyMiddleware(thunk));
```

Comment:
Try to add another reducer (say, a user reducer that could hold the user's name and shoe size) and use combineReducer to create a root reducer. Then, go back to all of your `mapStateToProps` statements and think about how you can optimize them. The purpose of mapStateToProps is to let you select the relevant parts of the state that your components need on an individual basis. You can imagine that, in a big app, there's gonna be a lot of stuff in our Redux state and passing it all to every connected component can slow things down.